enhancements/big-bundle-unpacking.md

```diff
@@ -0,0 +1,138 @@
+
+#### Determining Chunk Size
+
+Chunk-size will be determined by an option on the `catalog-operator` command, which if unset, directs the operator to probe the cluster for the resource storage limit. One potential probing strategy could be to create ConfigMaps of various sizes; akin to a "binary search" for the limit.
+
+### Risks and Mitigations
+
+Decompressing bundle content for each request may add significant time complexity. Some options for mitigating this are:
+
+- building an [MRU cache](https://en.wikipedia.org/wiki/Cache_replacement_policies#Most_recently_used_(MRU)) of bundle content
+- avoiding content compression entirely
+
+## Design Details
+
+### Version Skew Strategy
+
+#### The `Bundle` API and OLM
+
+The default aggregated API-server to service `Bundle` requests will be compiled and shipped w/ the rest of OLM; i.e. it will always match the version of OLM running on a given cluster.
+
+#### Unpack Jobs and OPM
+
+A small, statically linked binary will be generated from in-tree OLM source, for use in the unpack `Job`. This binary will vendor OPM packages to tightly couple content extraction logic to OLM version.
+
+#### Extraction Logic and Operator Bundle Mediatypes
+
+A given version of OLM will support unpackaging a fixed set of bundle mediatypes. If a bundle's mediatype is not within this set, an error will be surfaced in the status conditions of the respective `Bundle`. If OLM will drop support for a mediatype in an upcoming release, it will add a warning to the status conditions of all `Bundles` of said mediatype on a given cluster.
+
+## Alternatives
+
+- Instead of abstracting the unpackaging process behind a new API (`Bundle`), `opm` can be updated to partition bundle content across `ConfigMaps` directly. This would tightly couple OLM to one specific strategy.
```

Comment:
I think this is a good point and that this enhancement does a good job of abstracting away the underlying storage so that we can move to a more intuitive solution. 

This seems like a simple object storage problem. We can't call an s3 API from OLM but it would be nice if we could rely on an existing storage operator to provide some abstraction over volumes and we can simply put the manifests in a bucket. For example, minio provides this functionality.